packages/contracts-bedrock/scripts/DeployImplementations.s.sol

```diff
@@ -292,24 +302,42 @@ contract DeployImplementations is Script {
         });
     }
 
+    // Deploy and initialize the OPStackManager contract that is fronted by a proxy.
     function createOPSMContract(
         DeployImplementationsInput _dii,
         DeployImplementationsOutput,
-        OPStackManager.Blueprints memory blueprints
+        OPStackManager.Blueprints memory blueprints,
+        string memory release,
+        OPStackManager.ImplementationSetter[] memory setters
     )
         internal
         virtual
-        returns (OPStackManager opsm_)
+        returns (OPStackManager opsmProxy_)
     {
         SuperchainConfig superchainConfigProxy = _dii.superchainConfigProxy();
         ProtocolVersions protocolVersionsProxy = _dii.protocolVersionsProxy();
+        ProxyAdmin proxyAdmin = _dii.superchainProxyAdmin();
+        address proxyAdminOwner = proxyAdmin.owner();
 
-        vm.broadcast(msg.sender);
-        opsm_ = new OPStackManager({
-            _superchainConfig: superchainConfigProxy,
-            _protocolVersions: protocolVersionsProxy,
-            _blueprints: blueprints
-        });
+        // We broadcast as the proxyAdminOwner address due to the ProxyAdmin's `onlyOwner` modifier.
+        vm.startBroadcast(proxyAdminOwner);
+        Proxy proxy = new Proxy(address(proxyAdmin));
+        OPStackManager opsm = new OPStackManager();
+
+        proxyAdmin.upgradeAndCall(
```

Comment:
the multiple `upgradeAndCall` invocations is a weird pattern to me, but not sure what other way to do it if we make the ProxyAdmin the owner of the `OPStackManagerProxy`.

Reply:
I could possibly collapse these into two initializer functions but I played around with it a bit and could only get three.

Reply:
Resolving this comment based on latest implementation.